examples/server/server.cpp

```diff
@@ -2768,7 +2768,14 @@ static void append_to_generated_text_from_generated_token_probs(llama_server_con
 }
 
 std::function<void(int)> shutdown_handler;
-inline void signal_handler(int signal) { shutdown_handler(signal); }
+static bool is_terminating = false;
+inline void signal_handler(int signal) {
+    if (is_terminating) {
+        exit(0); // force exit the process, in case it hangs
+    }
+    is_terminating = true;
```

Comment:
It would be better to do this with std::atomic_flag test_and_set, since any thread can receive the signal.

Reply:
Thanks for suggestion, that was done in https://github.com/ggerganov/llama.cpp/pull/5734/commits/273d985271580bce0684889d007bbac7b2484e66